honeybee_schema/cli.py

```diff
@@ -48,47 +49,55 @@ def update_model(model_json, version, output_file):
-                importlib.import_module(mod, 'honeybee_schema.updater')
-                update_vers.append(ver)
-                update_functions.append('version_{}_{}_{}'.format(*ver))
-        func_mapper = {}
-        base_state = 'getattr(subpackage.{0}, "{0}")'
-        for func in update_functions:
-            statement = base_state.format(func)
-            namespace = {'subpackage': honeybee_schema.updater}
-            func_mapper[func] = eval(statement, namespace)
-
-        # execute the update modules on the Model dictionary to update it
-        update_vers, update_functions = zip(*sorted(zip(update_vers, update_functions)))
-        statement = 'update_funct(model_dict)'
-        for up_func in update_functions:
-            namespace = {'update_funct': func_mapper[up_func], 'model_dict': model_dict}
-            model_dict = eval(statement, namespace)
+        if model_version >= up_version:
+            # no point for updating. Normally we should assert here but by writing
+            # the same file to a new file we make this command more flexible for cases
+            # that we don't know the version for the input model and we want to update
+            # it just in case.
+            mv = '.'.join(map(str, model_version))
+            uv = '.'.join(map(str, up_version))
+            print(
+                f'The input file has a higher version ({mv}) than the tagert version'
+                f' ({uv}).', file=sys.stderr
+            )
+            output_file.write(json.dumps(model_dict))
+            # let's consider exporting the same file as success
+            sys.exit(0)
+
+        # get functions with a higher version than model_version
+        updaters = []
+
+        for sub_module in pkgutil.walk_packages(updater.__path__):
+            if not sub_module.name.startswith('version_'):
+                continue
+            module = importlib.import_module(f'honeybee_schema.updater.{sub_module.name}')
+            for name, func in getmembers(module, isfunction):
+                if not name.startswith('version_'):
+                    continue
+                func_version = tuple(int(v) for v in name.split('_')[1:])
+                if model_version > func_version:
+                    continue
+                updaters.append((func_version, func))
+
+        # sort updaters based on version
+        updaters = sorted(updaters, key=lambda x: x[0])
+
+        # update the dictionary
+        for version, up_func in updaters:
+            print(f'Updating to version {".".join(map(str, version))}', file=sys.stderr)
+            model_dict = up_func(model_dict)
 
         # update the model dictionary version and write it to log file
-        final_ver = up_version if version is not None else update_vers[-1]
-        model_dict['version'] = '.'.join((str(i) for i in final_ver))
+        if updaters:
+            model_dict['version'] = '.'.join((str(i) for i in updaters[-1][0]))
```

Comment:
The reason why I don't like this here is that you'll never be able to update a model HBJSON to a version of honeybee-schema that's not a breaking change, even if you specified a specific version you wanted to upgrade the model to with the `--version` option.

Given that most of the updates we make to honeybee-schema are not breaking changes, we really should respect the user input for the option here and assign it as the model['version'] instead of the version of the last breaking change.

Reply:
I changed it to work as you suggested. I'll merge it in if the tests pass.